### ISSUE TITLE
Exporting an encrypted database to unencrypted variety + SQL import issues

### ISSUE THREAD

**Issue by liroyvh** [sqlitebrowser/sqlitebrowser#2632]
Hi!

I have an encrypted database and need it stored in plain-text for conversion purposes.
When I open the .sqlite, I apparently do not have a way in DB Browser for SQLite to do this; it doesn't allow exporting to .sqlite. In order to work-around this problem, I have first used the browser to decrypt the database, then exported to SQL-format, then told it to re-import the .sql file. It prompts to create a new database and I choose yes.

However, upon importing the .sql file it has just created itself: it gives a syntax error, it encounters and unexpected bracket.
The export works fine without issue, the database can be browsed with no issue. But somehow, the export to .sql cannot be re-imported to a new unencrypted .sqlite file as there's something wrong with the .sql file. This means I'm kinda stuck, I can't export it to .sqlite and I can't import the .sql file in to a new .sqlite database.

What am I doing wrong and how can I simply save the encrypted .sqlite to an unencrypted .sqlite file? 
Also curious how it's possible that it exports fine but it cannot import the very database it has just exported itself haha. 
If it's easier to do this on the commandline that's fine as well, a quick Google didn't give me a satisfactory response how to achieve this, but maybe I google'd poorly. (Sorry if I have and this has already been addressed somewhere online! :))

**Comment by chrisjlocke** [comment]
>  I apparently do not have a way in DB Browser for SQLite to do this;

When you open the database, you can export the whole database to SQL statements, or individual tables to CSV (or even JSON now).

![image](https://user-images.githubusercontent.com/7872218/111436640-1f4cb980-86fa-11eb-8f17-956cc02a384a.png)

Is this what you're after?

**Comment by chrisjlocke** [comment]
> Also curious how it's possible that it exports fine but it cannot import the very database it has just exported itself

This could be due to how the database was exported, and what the records contain.
Lets take a rubbish example.  Lets say you have two fields - fieldA and fieldB.  When you export the database, it creates an SQL file which is just instructions to add that record.
`insert into tableA (fieldA,fieldB) values (1,2)`

Now lets assume someone was silly, and entered '1(' into fieldA instead.  This would break that instruction.
`insert into tableA (fieldA,fieldB) values (1(,2)`

DB4S usually spots this and doesn't do this, but sometimes, it can happen.  You can never code for everything that a user will do! 😁😁

Is the database private?  If you can share the exported file that you're trying to import (you could email it to me) then we could have a look at whats failing.  👍

**Comment by liroyvh** [comment]
@chrisjlocke Hi thank you for your response and my apologies for the delay. 

Yes what I'm trying to do is save the encrypted .sqlite as a plain-text .sqlite file so I can run some scripts against it. This doesn't seem to be possible. So I thought to be clever and export to .sql, then import the .sql back to a brand spanking new .sqlite.

Unfortunately it does export fine, but it refuses to import it hehe. So I'm kind of stuck trying to achieve my goal. :)  

Afraid the database in question contains my personal chat history in Signal, so not too keen on sharing that one 😄

**Comment by chrisjlocke** [comment]
OK, lets rewind a bit.  'run some scripts against it' - what are you trying to do?
As soon as you open the database in DB4S, its unencrypted, so you don't need to worry about exporting to import it back in again.  Depending on what you want your scripts to do, do you need just to look through a couple of columns, or are you editing data?  Lets assume you're just reading.
So we can create a view on the columns/rows you do want, and export that to CSV.  I can create a screencast on these steps.  (Actually just did, so yell back if you need help).  Click on 'File', 'Export' and 'Tables as CSV'.  Select the view, accept the defaults (theoretically) and you're away.

Also, 'run some scripts' - in what language?  If you know vb.net (or c#, or pretty much anything) you can talk to the database directly.  As its an encrypted database its a bit more difficult (the system.data.sqlite library doesn't support the same encryption method DB4S uses) so you have to use a paid-for library instead...

**Comment by liroyvh** [comment]
@chrisjlocke Hi Chris, sorry again for the delay. Ok so, a couple of people way smarter than me have written a couple of Python scripts that can backup Signal Messenger conversations and export them, for example to PDF. However, this was written when the database was a plain-text .sqlite file. It no longer is. So what I want to do is decrypt the .sqlite file and then store it as plain-text .sqlite again so I can run those scripts. 

But that idea was short-lived. I could open the database and decrypt it, but not store it to plain-text .sqlite. I could also export it to .sql; but importing it back (to a plain-text file :P) proved to be impossible as it errored out.

**Comment by chrisjlocke** [comment]
OK.  A 'plain text' .sqlite file isn't the same thing as a 'plain text' .sql file.  From DB4S, if you export to SQL, you'll get an .SQL file, not an .sqlite file.

An .sqlite file is the database itself.  It's not human-readable.  I would suspect the python script would require a .sqlite file, but that's just a guess.  It's far easier talking to a database than an .SQL file.
An .SQL file is a set of instructions necessary to 'rebuild' or create the .SQL file.  Its human-readable.  So from DB4S, you can export a database to .SQL, which means it creates a set of instructions (create table, insert into, etc) which is the database again.

When you open a database from DB4S, you are 'decrypting' it.  If you export it to SQL, it'll be in its decrypted form - you can't export it 'encrypted', as you can only export a set of instructions.
If you need a decrypted .sqlite file, you can do your export-import process, but you can also just remove the encryption in DB4S (ie, set no password) then 'save as'

![image](https://user-images.githubusercontent.com/7872218/113429456-76c47800-93d0-11eb-8638-6b3afb49a1dd.png)

Hope this makes sense with regards to .SQL and .sqlite files.  If you know exactly what type your scripts require, and you want to have another go, we can try again.  😃

**Comment by chrisjlocke** [comment]
I'll close this as there hasn't been any input for a while.
@liroyvh - feel free to comment when convenient and this can be re-opened.

**Comment by SirOsisOfTheLiver** [comment]
There is no Save Database As in db Browser for Sqlite 👍 
![image](https://user-images.githubusercontent.com/59989940/155857637-3cfd3766-0ea5-4980-aff8-8f7551368676.png)

I am using version 3.12.2

with sqlcipher version 4.4.3 communit
![ScreenshotDbBrowser](https://user-images.githubusercontent.com/59989940/155857686-5e2820a9-d8ad-43c5-8771-19e77240775c.png)
y

**Comment by chrisjlocke** [comment]
As SQLite is just one file, you can copy this file in Explorer and you have your new file.  Open this new file in DB4S, remove the encryption, and you have a new file without encryption.

**Comment by LiroyvH** [comment]
@chrisjlocke Sorry for not getting back about that. I ended up finally getting pysqlcipher3 working. (Its unmaintained and not very keen to compile under some circumstances.) This was pretty much what had earlier stopped me from simply backing changes in to the bash and/or python scripts and why I had to decrypt the database and then store it in a plain-text format instead. Now that the decryption can be done on the fly, the problem for me is sorted; but I unfortunately never got it to work with this tool. :) It was impossible to import it even though it would successfully export. 

Anyway, the goal I was trying to reach has been achieved: create PDF's of Signal Desktop chathistory hehe.
